feature/teaching/#45

itssisi opened the pull request:
Currently, the product displays all the TAs at once.

To facilitate easy of navigation for users, default course is set up using the TeachCommand.

This is so that the next time users log into the product, they can directly see the list of TAs teaching under them.

Closes #45.

itssisi committed:
Add TeachCommand

Currently, the product displays all the
in a single page.

To facilitate easy of navigation for
users, default course is set up using
TeachCommand command.

This is so that the next time users log into
the product, they can directly see the list
of TAs teaching under them.

itssisi committed:
Merge branch 'feature/TeachCommand/#45' into feature/teaching/#45

itssisi committed:
Add Javadocs for TeachCommand

Some Javadocs for TeachCommand are missing.

Hence, for clarity of descriptions,
Javadocs are added.

lshaoqin commented on src/main/java/seedu/address/model/Model.java:
```diff
@@ -44,6 +45,17 @@ public interface Model {
      */
     void setAddressBookFilePath(Path addressBookFilePath);
 
+    /**
+     * Returns the module the user is teaching.
+     *
+     */
+    Mod getTeaching();
+
+    /**
+     * Sets the module the user is teaching.
+     */
+    void setTeaching(Mod module);
+
```
I don't think these methods should be part of the Model interface, as they might not be implemented in other models if we choose to extend it.

itssisi replied:
It is necessary for the `Logic` class to fetch the default course. So, it is still there.

lshaoqin commented on src/main/java/seedu/address/model/UserPrefs.java:
```diff
@@ -16,6 +17,7 @@ public class UserPrefs implements ReadOnlyUserPrefs {
     private GuiSettings guiSettings = new GuiSettings();
     private Path addressBookFilePath = Paths.get("data" , "addressbook.json");
 
+    private Mod teaching;
```
How does this field influence startup behavior at the moment?

itssisi replied:
The title of the window changes (refer to screenshot). The list of TAs are filtered according to the default course automatically.

lshaoqin reviewed (commented):
I need some clarification as to what the intended functionality of the comment is.

Code is clean overall but testcases are needed. In addition, the refactoring of Mod into Course would affect this PR greatly. Let's wait for #40 to be resolved before continuing work on this.

itssisi committed:
Merge branch 'master' into feature/teaching/#45

itssisi committed:
Rename Mod Course

This needs to be updated due to upstream refactoring

itssisi committed:
Add DefaultModule Box

itssisi committed:
Merge branch 'master' into feature/teaching/#45

itssisi committed:
Adds the behaviour of teachCommand

There were some problems with the behaviour of the teachCommand, such as not updating the json file and displaying the course name instead of the course code.

The userPref behaviour has been updated to show the changes when teachCommand is executed.

lshaoqin commented on src/main/java/seedu/address/logic/commands/TeachCommand.java:
```diff
@@ -0,0 +1,69 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sets up default course users are teaching.\n"
+            + "Parameters: "
+            + PREFIX_COURSE + "COURSE \n"
```
let's standardise this as COURSE_CODE

itssisi replied:
Bug number 1 is related to @raynertjx. 

In our current feature, once the default course is set up, it is not meant to be reset from the product. We can do it manually by deleting the default course at the json file.

itssisi replied:
> let's standardise this as COURSE_CODE

fixed

lshaoqin reviewed (changes requested):
Good job! Nice reuse of TeachingCoursePredicate here.

Two bugs that I found:
1) After setting the course to teach, the find command can still find TAs from outside the course. Not sure if this is intended behaviour.
2) `find n/` will result in `java.lang.reflect.InvocationTargetException` but no error message is shown to the user in the app. Tagging @raynertjx for this.

Also, perhaps there should be a way to reset the addressbook to having no default course?

itssisi committed:
Fixes naming convention

Fixes COURSE to be COURSE_CODE in TeachCommand
